rocketpy/motors/Motor.py

```diff
@@ -145,88 +155,169 @@ def __init__(
                 thrustSource = points
-                self.burnOutTime = points[-1][0]
 
-        # Create thrust function
+        # Evaluate raw thrust source
+        self.interpolate = interpolationMethod
+        self.thrustSource = thrustSource
         self.thrust = Function(
             thrustSource, "Time (s)", "Thrust (N)", self.interpolate, "zero"
         )
-        if callable(thrustSource) or isinstance(thrustSource, (int, float)):
-            self.thrust.setDiscrete(0, burnOut, 50, self.interpolate, "zero")
 
-        # Reshape curve
+        # Handle burn_time input
+        self.burn_time = burn_time
+
+        if callable(self.thrust.source):
+            self.thrust.setDiscrete(*self.burn_time, 50, self.interpolate, "zero")
+
+        # Reshape thrustSource if needed
         if reshapeThrustCurve:
+            # Overwrites burn_time and thrust
             self.reshapeThrustCurve(*reshapeThrustCurve)
 
+        # Post process thrust
+        self.thrust = self.clipThrust()
+
+        # Auxiliary quantities
+        self.burnDuration = self.burn_time[1] - self.burn_time[0]
+        self.burnOutTime = self.burn_time[1]
+
+        # Define motor attributes
+        self.nozzleRadius = nozzleRadius
+        self.nozzlePosition = nozzlePosition
+
         # Compute thrust metrics
-        self.maxThrust = np.amax(self.thrust.source[:, 1])
-        maxThrustIndex = np.argmax(self.thrust.source[:, 1])
+        self.maxThrust = np.amax(self.thrust.yArray)
+        maxThrustIndex = np.argmax(self.thrust.yArray)
         self.maxThrustTime = self.thrust.source[maxThrustIndex, 0]
-        self.averageThrust = self.totalImpulse / self.burnOutTime
+        self.averageThrust = self.totalImpulse / self.burnDuration
+
+    @property
+    def burn_time(self):
+        """Burn time range in seconds.
+
+        Returns
+        -------
+        tuple
+            Burn time range in seconds.
+        """
+        return self._burn_time
+
+    @burn_time.setter
+    def burn_time(self, burn_time):
+        """Sets burn time range in seconds.
+
+        Parameters
+        ----------
+        burn_time : float or two position array_like
+        """
+        if isinstance(burn_time, (int, float)):
+            self._burn_time = (0, burn_time)
+        elif isinstance(burn_time, (list, tuple)):
+            if len(burn_time) == 1:
+                self._burn_time = (0, burn_time[0])
+            elif len(burn_time) == 2:
+                self._burn_time = burn_time
+            else:
+                raise ValueError("burn_time must be a list or tuple of length 1 or 2.")
+        elif callable(self.thrust.source):
+            raise ValueError(
+                "When using a float or callable as thrust source a burn_time "
+                "range must be specified."
+            )
+        else:
+            self._burn_time = (self.thrust.xArray[0], self.thrust.xArray[-1])
+
+    def clipThrust(self):
```

Comment:
Missing docstrings in this function

Reply:
Solved by [081f0de](https://github.com/RocketPy-Team/RocketPy/pull/337/commits/081f0de83c750b1e936a149b49759e5f97e8ea96).